tests/tests/relayer.rs

```diff
@@ -252,6 +263,41 @@ async fn messages_are_spendable_after_relayer_is_synced() {
     eth_node_handle.shutdown.send(()).unwrap();
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn can_find_failed_relayed_tx() {
+    let mut db = CombinedDatabase::in_memory();
+    let id = [1; 32].into();
+    let block_height: BlockHeight = 999.into();
+    let block_time = Tai64::UNIX_EPOCH;
+    let failure = "lolz".to_string();
+
+    // given
+    let status = FuelRelayedTransactionStatus::Failed {
+        block_height,
+        block_time,
+        failure: failure.clone(),
+    };
+    db.off_chain_mut()
```

Comment:
Should we have an integration test involving mocked events from the relayer? That way we know the graphql worker task properly updates the offchain table.

Reply:
We do have a UT that is kinda doing that: `run__relayed_transaction_events_are_added_to_storage`.

I'm trying to understand the full flow you want. Are you looking to have the 
1. relayer send a _bad_ transaction that is then 
2. included in block production 
3. the worker would then see the failure event in the block and record it in the DB and
4. then we would query it with the client?

I think we have coverage of all these sections from the UT perspective, but you're right that we don't see them all working together in one flow

Reply:
Yeah after reviewing some more, it does seem like we have full e2e coverage split across various unit tests. We can defer full e2e testing to the bridge integration if we feel confident in what we already have here.

Reply:
That sounds good to me.